### PR TITLE
Fix: remaining cycles calculation

### PR DESCRIPTION
**Pull request by maikelordaz**
The remaining cycles were calculated according old requirements, when we did not had the need to check anything betwen the end of a contribution period and the start of the next cycle. But this affect the calculation of undercollaterization to freeze money pots. A user can be affected in the next way

<details>
<summary>Bug case</summary>

- A turn group is created with four participants
- User A joins third position
- User A defaults first cycle and pay in the second one
- Third cycle starts and ends the contribution period
- The SC check the frozen money pot as if there are two cycles left but it is only one
- As user A defaulted once and he payed less security amount, he now have enough to cover the cycle left
- User A does not have enough to cover two cycles (as SC think is left)
- User A's money pot is frozen until the last cycle starts, or ETH price increase enough

</details>

<details>
<summary>Fix</summary>

Now the remaining cycles (RC) are calculated as

- If the fund is initializing, the RC is equal to participants number
- If the fund is accepting contributions or awarding the beneficiary, RC it is calculated as current deployed code
- If the fund already award the beneficiary and it is waiting for the next cycle to start, RC is equal to the one in the previous point, minus 1

</details>

### CODE REVIEW

File: contracts/facets/GettersFacet.sol [hunk 1]
```diff
@@ -716,7 +716,19 @@ contract GettersFacet is IGetters {
...
+
+        if (fund.currentState == LibFundStorage.FundStates.InitializingFund) {
+            remainingCycles = term.totalParticipants;
+        } else if (
+            fund.currentState == LibFundStorage.FundStates.AcceptingContributions ||
+            fund.currentState == LibFundStorage.FundStates.AwardingBeneficiary
+        ) {
+            remainingCycles = getRemainingCycles(termId);
+        } else if (fund.currentState == LibFundStorage.FundStates.CycleOngoing) {
+            remainingCycles = getRemainingCycles(termId) - 1;
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol:729]: And this does not cause an underflow on the last cycle?

Comment (maikelordaz) [reply contracts/facets/GettersFacet.sol:729]: No it will not, when the cycle is accepting contributions or awarding beneficiary it will calculate the remaining cycles as 1. When the funding period is over, as it is the last cycle the state will be FundClosed, as you can see that one is not one of the edge cases so the remainingCycles variable will remain as 0 because it was never initialized. But now that you ask, think I better add that if statement for better readability

Comment (maikelordaz) [reply contracts/facets/GettersFacet.sol:729]: @MohammedTaka in the last commits I add a new edge case, the functionality remains the same but think is best for readability and future manual reviews. Also added a test case to ensure it does not underflow in the last cycle

**Review by elallamTaka — APPROVED** [review]

**Review by MohammedTaka — APPROVED** [review]
Double checked the logic, we can indeed reduce the remaining cycles by 1 if the state is ongoing.
